lib/upgrade/etcd/conversionv1v3/hostendpoint_test.go

```diff
@@ -0,0 +1,153 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conversionv1v3
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	apiv1 "github.com/projectcalico/libcalico-go/lib/apis/v1"
+	"github.com/projectcalico/libcalico-go/lib/apis/v1/unversioned"
+	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	//"github.com/projectcalico/libcalico-go/lib/ipip"
+	cnet "github.com/projectcalico/libcalico-go/lib/net"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"github.com/projectcalico/libcalico-go/lib/numorstring"
+)
+
+const hepName = "my-hep"
+const nodeName = "my-node"
+const profileName = "my-prof"
+const ifName = "eth3"
+const portName = "my-port"
+const portNum = 12345
+const portProtocol = "tcp"
+const ip = "192.168.1.1"
+
+var hepTable = []struct {
+	description string
+	v1API unversioned.Resource
+	v1KVP *model.KVPair
+	v3API apiv3.HostEndpoint
+}{
+	{
+		description: "Super basic hep",
+		v1API: apiv1.HostEndpoint{
+			Metadata: apiv1.HostEndpointMetadata{
+				Name: hepName,
+			},
+			Spec: apiv1.HostEndpointSpec{},
+		},
+		v1KVP: &model.KVPair{
+			Key: model.HostEndpointKey{
+				EndpointID: hepName,
+			},
+			Value: &model.HostEndpoint{
+			},
+		},
+		v3API: apiv3.HostEndpoint{
+			ObjectMeta: v1.ObjectMeta{
+				Name: hepName,
+			},
+			Spec: apiv3.HostEndpointSpec{
+				ExpectedIPs: []string{},
+			},
+		},
+	},
+	{
+		description: "fully loaded hep",
+		v1API: apiv1.HostEndpoint{
+			Metadata: apiv1.HostEndpointMetadata{
+				Name: hepName,
+				Node: nodeName,
+			},
+			Spec: apiv1.HostEndpointSpec{
+				ExpectedIPs: []cnet.IP{cnet.MustParseIP(ip)},
+				Profiles: []string{profileName},
+				InterfaceName: ifName,
+				Ports: []apiv1.EndpointPort{
+					{
+						Name:     portName,
+						Protocol: numorstring.ProtocolFromStringV1(portProtocol),
+						Port: portNum,
+					},
+				},
+			},
+		},
+		v1KVP: &model.KVPair{
+			Key: model.HostEndpointKey{
+				EndpointID: hepName,
+				Hostname: nodeName,
+			},
+			Value: &model.HostEndpoint{
+				Name: ifName,
+				Ports: []model.EndpointPort{
+					{
+						Name: portName,
+						Port: portNum,
+						Protocol: numorstring.ProtocolFromStringV1(portProtocol),
+					},
+				},
+				ProfileIDs: []string{profileName},
+				ExpectedIPv4Addrs: []cnet.IP{cnet.MustParseIP(ip)},
+			},
+		},
+		v3API: apiv3.HostEndpoint{
+			ObjectMeta: v1.ObjectMeta{
+				Name: hepName,
+			},
+			Spec: apiv3.HostEndpointSpec{
+				InterfaceName: ifName,
+				Node: nodeName,
+				Profiles: []string{profileName},
+				ExpectedIPs: []string{ip},
+				Ports: []apiv3.EndpointPort{
+					{
+						Protocol: numorstring.ProtocolFromStringV1(portProtocol),
```

Comment:
I'm guessing I need to convert from `ProtocolFromStringV1` to `ProtocolFromString`. Could use some advice here

Reply:
yeah we'll need to handle that in the conversion logic, so it'll be ProtocolFromString here when that's done